libavcodec/vvc/vvcdsp_template.c

```diff
@@ -93,26 +94,164 @@ static void FUNC(transform_bdpcm)(int *coeffs, const int width, const int height
     }
 }
 
-static void FUNC(ff_vvc_itx_dsp_init)(VVCItxDSPContext *const itx)
-{
-#define VVC_ITX(TYPE, type, s)                                                  \
-        itx->itx[TYPE][TX_SIZE_##s]      = itx_##type##_##s;                    \
+#define ITX_COMMON_SIZES(TYPE_H, type_h, TYPE_V, type_v)                        \
+    ITX_1D_V(TYPE_H, type_h, TYPE_V, type_v, 1, 4);                             \
+    ITX_1D_V(TYPE_H, type_h, TYPE_V, type_v, 1, 8);                             \
+    ITX_1D_V(TYPE_H, type_h, TYPE_V, type_v, 1, 16);                            \
+    ITX_1D_V(TYPE_H, type_h, TYPE_V, type_v, 1, 32);                            \
+    ITX_1D_H(TYPE_H, type_h, TYPE_V, type_v, 4, 1);                             \
+    ITX_2D(TYPE_H, type_h, TYPE_V, type_v, 4, 4);                               \
+    ITX_2D(TYPE_H, type_h, TYPE_V, type_v, 4, 8);                               \
+    ITX_2D(TYPE_H, type_h, TYPE_V, type_v, 4, 16);                              \
+    ITX_2D(TYPE_H, type_h, TYPE_V, type_v, 4, 32);                              \
+    ITX_1D_H(TYPE_H, type_h, TYPE_V, type_v, 8, 1);                             \
+    ITX_2D(TYPE_H, type_h, TYPE_V, type_v, 8, 4);                               \
+    ITX_2D(TYPE_H, type_h, TYPE_V, type_v, 8, 8);                               \
+    ITX_2D(TYPE_H, type_h, TYPE_V, type_v, 8, 16);                              \
+    ITX_2D(TYPE_H, type_h, TYPE_V, type_v, 8, 32);                              \
+    ITX_1D_H(TYPE_H, type_h, TYPE_V, type_v, 16, 1);                            \
+    ITX_2D(TYPE_H, type_h, TYPE_V, type_v, 16, 4);                              \
+    ITX_2D(TYPE_H, type_h, TYPE_V, type_v, 16, 8);                              \
+    ITX_2D(TYPE_H, type_h, TYPE_V, type_v, 16, 16);                             \
+    ITX_2D(TYPE_H, type_h, TYPE_V, type_v, 16, 32);                             \
+    ITX_1D_H(TYPE_H, type_h, TYPE_V, type_v, 32, 1);                            \
+    ITX_2D(TYPE_H, type_h, TYPE_V, type_v, 32, 4);                              \
+    ITX_2D(TYPE_H, type_h, TYPE_V, type_v, 32, 8);                              \
+    ITX_2D(TYPE_H, type_h, TYPE_V, type_v, 32, 16);                             \
+    ITX_2D(TYPE_H, type_h, TYPE_V, type_v, 32, 32);
 
-#define VVC_ITX_COMMON(TYPE, type)                                              \
-        VVC_ITX(TYPE, type, 4);                                                 \
-        VVC_ITX(TYPE, type, 8);                                                 \
-        VVC_ITX(TYPE, type, 16);                                                \
-        VVC_ITX(TYPE, type, 32);
+#define ITX \
+    ITX_COMMON_SIZES(DCT2, dct2, DCT2, dct2);                                   \
```

Comment:
maybe you can define 
```
#define ITX_2D_WX(TYPE_H, type_h, TYPE_V, type_v, W) \
    ITX_2D_(TYPE_H, type_h, TYPE_V, type_v, W, 2); \
    ITX_2D_(TYPE_H, type_h, TYPE_V, type_v, W, 4）; \
    ..
```

then you can do 

```
#define ITX_COMMON_SIZES(DCT2, dct2, DCT2, dct2) \
    ITX_2D_WX(DCT2, dct2, DCT2, dct2, 2);
    ....

```

Reply:
Like this:
```diff
diff --git a/libavcodec/vvc/vvcdsp_template.c b/libavcodec/vvc/vvcdsp_template.c
index 4cba360411..cf9fd2f69e 100644
--- a/libavcodec/vvc/vvcdsp_template.c
+++ b/libavcodec/vvc/vvcdsp_template.c
@@ -94,31 +94,19 @@ static void FUNC(transform_bdpcm)(int *coeffs, const int width, const int height
     }
 }
 
+#define ITX_COMMON_WIDTHS(TYPE_H, type_h, TYPE_V, type_v, height)               \
+    ITX_1D_V(TYPE_H, type_h, TYPE_V, type_v, 1, height)                         \
+    ITX_1D_V(TYPE_H, type_h, TYPE_V, type_v, 4, height)                         \
+    ITX_1D_V(TYPE_H, type_h, TYPE_V, type_v, 8, height)                         \
+    ITX_1D_V(TYPE_H, type_h, TYPE_V, type_v, 16, height)                        \
+    ITX_1D_V(TYPE_H, type_h, TYPE_V, type_v, 32, height)                        \
+
 #define ITX_COMMON_SIZES(TYPE_H, type_h, TYPE_V, type_v)                        \
-    ITX_1D_V(TYPE_H, type_h, TYPE_V, type_v, 1, 4);                             \
-    ITX_1D_V(TYPE_H, type_h, TYPE_V, type_v, 1, 8);                             \
-    ITX_1D_V(TYPE_H, type_h, TYPE_V, type_v, 1, 16);                            \
-    ITX_1D_V(TYPE_H, type_h, TYPE_V, type_v, 1, 32);                            \
-    ITX_1D_H(TYPE_H, type_h, TYPE_V, type_v, 4, 1);                             \
-    ITX_2D(TYPE_H, type_h, TYPE_V, type_v, 4, 4);                               \
-    ITX_2D(TYPE_H, type_h, TYPE_V, type_v, 4, 8);                               \
-    ITX_2D(TYPE_H, type_h, TYPE_V, type_v, 4, 16);                              \
-    ITX_2D(TYPE_H, type_h, TYPE_V, type_v, 4, 32);                              \
-    ITX_1D_H(TYPE_H, type_h, TYPE_V, type_v, 8, 1);                             \
-    ITX_2D(TYPE_H, type_h, TYPE_V, type_v, 8, 4);                               \
-    ITX_2D(TYPE_H, type_h, TYPE_V, type_v, 8, 8);                               \
-    ITX_2D(TYPE_H, type_h, TYPE_V, type_v, 8, 16);                              \
-    ITX_2D(TYPE_H, type_h, TYPE_V, type_v, 8, 32);                              \
-    ITX_1D_H(TYPE_H, type_h, TYPE_V, type_v, 16, 1);                            \
-    ITX_2D(TYPE_H, type_h, TYPE_V, type_v, 16, 4);                              \
-    ITX_2D(TYPE_H, type_h, TYPE_V, type_v, 16, 8);                              \
-    ITX_2D(TYPE_H, type_h, TYPE_V, type_v, 16, 16);                             \
-    ITX_2D(TYPE_H, type_h, TYPE_V, type_v, 16, 32);                             \
-    ITX_1D_H(TYPE_H, type_h, TYPE_V, type_v, 32, 1);                            \
-    ITX_2D(TYPE_H, type_h, TYPE_V, type_v, 32, 4);                              \
-    ITX_2D(TYPE_H, type_h, TYPE_V, type_v, 32, 8);                              \
-    ITX_2D(TYPE_H, type_h, TYPE_V, type_v, 32, 16);                             \
-    ITX_2D(TYPE_H, type_h, TYPE_V, type_v, 32, 32);
+    ITX_COMMON_WIDTHS(TYPE_h, type_h, TYPE_V, type_v, 1)                        \
+    ITX_COMMON_WIDTHS(TYPE_h, type_h, TYPE_V, type_v, 4)                        \
+    ITX_COMMON_WIDTHS(TYPE_h, type_h, TYPE_V, type_v, 8)                        \
+    ITX_COMMON_WIDTHS(TYPE_h, type_h, TYPE_V, type_v, 16)                       \
+    ITX_COMMON_WIDTHS(TYPE_h, type_h, TYPE_V, type_v, 32)
 
 #define ITX \
     ITX_COMMON_SIZES(DCT2, dct2, DCT2, dct2);                                   \
```
do you mean?

This is slightly less readable imo and it only saves 12 lines, but I am happy to add it if you think it's better. 

Sidenote: the patch above actually fails to compile at the moment as I have not implemented the size-1 1-D transforms (i.e. for transform size 1x1). These an edge case and are not used in any of the conformance bitstreams, but I think they may actually be valid per the spec and should be implemented.